index.js

```diff
@@ -8,11 +8,100 @@
 'use strict';
 
 var fs = require('fs');
+var path = require('path');
+
+/**
+ * Check if the given `filepath` exists.
+ *
+ * ```js
+ * var res = exists('package.json');
+ * console.log(res);
+ * //=> "package.json"
+ *
+ * var res = exists('fake-file.json');
+ * console.log(res)
+ * //=> false
+ * ```
+ *
+ * @name exists
+ * @param  {String} `filepath` filepath to check for.
+ * @return {String|Boolean} Returns the found filepath if it exists, otherwise returns `false`.
+ * @api public
+ */
 
 module.exports = function(filepath) {
+  if (!filepath || (typeof filepath !== 'string')) {
+    return false;
+  }
+
   try {
     (fs.accessSync || fs.statSync)(filepath);
-    return true;
+    return filepath;
   } catch (err) {}
+
+  if (process.platform === 'linux') {
```

Comment:
okay, let do more research then. It's my understanding that only linux has this specific issue with `fs.statSync`. If anyone knows where this is documented that would be helpful